src/Coordinate.java

```diff
@@ -28,19 +34,26 @@ public String call() throws Exception {
 	@Override
 	public void attach(Ship observer) {
 		// TODO
+		super.attach(observer);
 	}
 
 	// TODO
 	// return if the coordinate has been called or not
 	//
 	public boolean called() {
-		return false;
+		if (this.called = false){
+			return false;
+			
+		} else {
+			return true;
+		}
+
 	}
 
 	// TODO
 	// is there a ship on this coordinate?
 	//
 	public boolean hasShip() {
-		return false;
+		return hasShip;
```

Comment:
Notice that this project utilizes the `this` keyword in contexts where it's not always needed. It's best to be consistent throughout the entire codebase.